Icinga2 fails to build on Windows

## Current Behavior
While trying to build the current master branch on Windows I came across this error: 

```
C:\boost_1_65_1\boost/mpl/at.hpp(44): error C2903: "apply": Symbol ist weder eine Klassen-Vorlage noch eine Funktions-Vorlage. [C:\Users\Michael\source\repos\icinga2\
lib\base\base.vcxproj]
  C:\Users\Michael\source\repos\icinga2\lib\base/functionwrapper.hpp(143): note: Siehe Verweis auf die Klasse Vorlage-Instanziierung "boost::mpl::at_c<int,1>", die ko
  mpiliert wird.
  C:\Users\Michael\source\repos\icinga2\lib\base/array-script.cpp(254): note: Siehe Verweis auf die Instanziierung der gerade kompilierten Funktions-Vorlage "icinga::
  Function::Function<double(__cdecl *)(void)>(const icinga::String &,F,const std::vector<icinga::String,std::allocator<_Ty>> &,bool,bool)".
          with
          [
              F=double (__cdecl *)(void),
              _Ty=icinga::String
          ]
C:\boost_1_65_1\boost/mpl/at.hpp(44): error C3770: "unknown-type": ist keine gültige Basisklasse [C:\Users\Michael\source\repos\icinga2\lib\base\base.vcxproj]
```

--- a/lib/base/functionwrapper.hpp
+++ b/lib/base/functionwrapper.hpp
@@ -38,11 +38,11 @@ template<typename FuncType>
 typename std::enable_if<
     std::is_class<FuncType>::value &&
     std::is_same<typename boost::function_types::result_type<decltype(&FuncType::operator())>::type, Value>::value &&
-    std::is_same<typename boost::mpl::at_c<boost::function_types::parameter_types<decltype(&FuncType::operator())>, 1>::type, const std::vector<Value>&>::value &&
-    boost::function_types::function_arity<decltype(&FuncType::operator())>::value == 2,
+	boost::function_types::function_arity<decltype(&FuncType::operator())>::value == 2,
     std::function<Value (const std::vector<Value>&)>>::type
 WrapFunction(FuncType function)
 {
+	static_assert(std::is_same<typename boost::mpl::at_c<typename boost::function_types::parameter_types<decltype(&FuncType::operator())>, 1>::type, const std::vector<Value>&>::value, "Argument type must be const std::vector<Value>");
 	return function;
 }
 
@@ -140,11 +140,12 @@ template<typename FuncType>
 typename std::enable_if<
     std::is_class<FuncType>::value &&
     !(std::is_same<typename boost::function_types::result_type<decltype(&FuncType::operator())>::type, Value>::value &&
-    std::is_same<typename boost::mpl::at_c<boost::function_types::parameter_types<decltype(&FuncType::operator())>, 1>::type, const std::vector<Value>&>::value &&
-    boost::function_types::function_arity<decltype(&FuncType::operator())>::value == 2),
+	boost::function_types::function_arity<decltype(&FuncType::operator())>::value == 2),
     std::function<Value (const std::vector<Value>&)>>::type
 WrapFunction(FuncType function)
 {
+	static_assert(!std::is_same<typename boost::mpl::at_c<typename boost::function_types::parameter_types<decltype(&FuncType::operator())>, 1>::type, const std::vector<Value>&>::value, "Argument type must be const std::vector<Value>");
+
 	using FuncTypeInvoker = decltype(&FuncType::operator());
 
 	return [function](const std::vector<Value>& arguments) {

